packages/style-engine/class-wp-style-engine-css-rule.php

```diff
@@ -107,19 +139,28 @@ public function get_selector() {
 		 * @return string
 		 */
 		public function get_css( $should_prettify = false, $indent_count = 0 ) {
-			$rule_indent         = $should_prettify ? str_repeat( "\t", $indent_count ) : '';
-			$declarations_indent = $should_prettify ? $indent_count + 1 : 0;
-			$suffix              = $should_prettify ? "\n" : '';
-			$spacer              = $should_prettify ? ' ' : '';
+			$rule_indent                = $should_prettify ? str_repeat( "\t", $indent_count ) : '';
+			$nested_rule_indent         = $should_prettify ? str_repeat( "\t", $indent_count + 1 ) : '';
+			$declarations_indent        = $should_prettify ? $indent_count + 1 : 0;
+			$nested_declarations_indent = $should_prettify ? $indent_count + 2 : 0;
+			$suffix                     = $should_prettify ? "\n" : '';
+			$spacer                     = $should_prettify ? ' ' : '';
 			// Trims any multiple selectors strings.
-			$selector         = $should_prettify ? implode( ',', array_map( 'trim', explode( ',', $this->get_selector() ) ) ) : $this->get_selector();
-			$selector         = $should_prettify ? str_replace( array( ',' ), ",\n", $selector ) : $selector;
-			$css_declarations = $this->declarations->get_declarations_string( $should_prettify, $declarations_indent );
+			$selector                = $should_prettify ? implode( ',', array_map( 'trim', explode( ',', $this->get_selector() ) ) ) : $this->get_selector();
+			$selector                = $should_prettify ? str_replace( array( ',' ), ",\n", $selector ) : $selector;
+			$css_declarations        = $this->declarations->get_declarations_string( $should_prettify, $declarations_indent );
+			$nested_css_declarations = $this->declarations->get_declarations_string( $should_prettify, $nested_declarations_indent );
```

Comment:
Tiny nit, and doesn't affect the API changes here, so not a blocker: Why is `get_declarations_string` called twice? Is there ever a time we'd need to have access to both `$css_declarations` and `$nested_css_declarations`?

Just wondering if something like the following could avoid the extra call:

```
$at_rule                 = $this->get_at_rule();
$has_at_rule             = ! empty( $at_rule );

$nested_css_declarations = $this->declarations->get_declarations_string(
    $should_prettify,
    $has_at_rule ? $nested_declarations_indent : $declarations_indent
);
```

Reply:
Oooh yeah that's a good idea! There's no need to access both at the same time.